### PR TITLE
MODCON-26 added tenant unique code. error handling updated

### PR DESCRIPTION
**Pull request by grabsefx**
## Purpose
https://issues.folio.org/browse/MODCON-26

## Approach
* added tenant unique code. error handling updated

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by grabsefx** [comment]
Error response example:
```
{
  "errors": [
    {
      "message": "code The code must be alphanumeric.",
      "type": "-1",
      "code": "TenantEntityValidationError"
    },
    {
      "message": "code Invalid Code length: Must be of 3 alphanumeric characters",
      "type": "-1",
      "code": "TenantEntityValidationError"
    }
  ]
}
```

File: src/main/resources/db/changelog/changes/create-tenant-table.xml [hunk 1]
```diff
@@ -18,5 +18,12 @@
       </column>
     </createTable>
   </changeSet>
+  <changeSet id="MOCON-26@add-code-unique-field" author="siarhei_hrabko@epam.com">
```
Comment (grabsefx) [inline src/main/resources/db/changelog/changes/create-tenant-table.xml:21]: could be moved to the previous change set. to discuss

File: src/main/java/org/folio/consortia/controller/ErrorHandlingController.java [hunk 2]
```diff
@@ -78,25 +79,22 @@ public Errors handleValidationErrors(Exception e) {
...
+  public Errors handleConstraintViolation(ConstraintViolationException ex) {
     log.error("Handle constraint violation", ex);
 
-    // Extract the error message and validation errors from the ConstraintViolationException
-    List<String> validationErrors = ex.getConstraintViolations().stream()
-      .map(violation -> violation.getPropertyPath() + " " + violation.getMessage())
+    List<Error> errorList = ex.getConstraintViolations()
+      .stream()
+      .map(violation -> {
+        var customCode = violation.getRootBeanClass() != null ? violation.getRootBeanClass().getSimpleName() : EMPTY + "ValidationError";
```
[the remark this answers is not in the document]
Comment (SerhiiNosko) [reply src/main/java/org/folio/consortia/controller/ErrorHandlingController.java]: added braces to improve readability

**Comment by azizbekxm** [comment]
Karate test result:
image

**Review by SerhiiNosko — APPROVED** [review]

**Review by azizbekxm — APPROVED** [review]

**Review by SinghAdes — APPROVED** [review]

**Comment by azizbekxm** [comment]
wow, so many approves

**Pull request merged**